packages/@lwc/engine/src/framework/vm.ts

```diff
@@ -412,7 +412,7 @@ function runDisconnectedCallback(vm: VM) {
         // it will be re-rendered because we are disconnecting the reactivity
         // linking, so mutations are not automatically reflected on the state
         // of disconnected components.
-        markComponentAsDirty(vm);
+        vm.isDirty = true;
```

Comment:
What was this change? Was it a bug fix?

Reply:
Yes. 
1. When a parent is being rehydrated, the old children in the parent's shadow tree are disconnected and marked as dirty(so that in the future if the child is reused, we know it has to be rehydrated).
2. During runDisconnectedCallback() invocation for the child, we know the child is not in a rendering phase but the parent is. 
3. To mark the child as dirty, we were calling markComponentAsDirty(), but that has an assertion to verify that we are not in a rendering phase(to avoid infinite loop).
4. So we short circuit that and directly mark the child's vm as dirty.